src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__BIONIC__))
```

Comment:
Then you also need to add cast to T * to the second parameter like this:
```C++
__atomic_load((T const *)pt, (T *)&val, __ATOMIC_ACQUIRE);
```

The problem is that the VolatileLoad function is generic and so if it is passed a pointer that is volatile, it derives the type `T` as being volatile. But the signature of the __atomic_load has both parameters as non-volatile pointers and so the compiler complains.
I was slightly wrong with the prototype in my previous response, it is defined rather as
```C++
void __atomic_load (type *ptr, type *ret, int memmodel)
```
The likely reason why it doesn't complain when building on Linux is probably due to a different version of clang that is used for android and Linux.